### ISSUE TITLE
Softmax & Probabilities

### ISSUE THREAD

**Issue by ingo** [DuckieOverload/transferlearningsophon#4]
## Goal
Understand how model outputs become probabilities.

## Watch
Neural Networks Part 5: ArgMax and SoftMax
https://www.youtube.com/watch?v=KpKog-L9veg

## Tasks
- [ ] Watch the video
- [ ] Answer:
  - Why do softmax outputs always sum to 1?
  - What do the individual softmax values represent?
  - Why can’t we just use ArgMax during training?
  - What is the difference between choosing the largest score and computing probabilities?
  - In a classification model, when do we use softmax and when do we use argmax?

## Repo Pairing
- Find where softmax or cross-entropy loss is used
- Inspect final model outputs

## Done When
You can explain what a softmax probability means.

**Comment by DuckieOverload** [comment]
1. Softmax sums to 1 because of the two-step normalization process, where everything is raised to the power of e so everything is positive, then divides each of those positive values by the sum of all the others combined. This creates a mathematical ratio where the individual parts must always add up to 100%, or 1.
2. Each individual value represents the probability or confidence level the model assigns to a specific class. If a category is 0.83 for example, it is saying that it is 0.83% that the input is that category.
3. We can't use it in training because it is not differentiable. Models use the slopes of lines to train (which requires differentiation for slopes at specific points). However, since ArgMax is perfectly flat, it provides no useful data to the model in training.
4. The difference between choosing the largest score and computing probabilities is the depth of the data you receive. ArgMax tells you which is best, but does not tell you how much closer anyone else was. SoftMax does.
5. Softmax is used in training, and ArgMax is used when we are all done and need no more training.
